test/docker_run_tests.sh

```diff
@@ -0,0 +1,208 @@
+#!/bin/sh
+
+# Script will self manage errors:
+set +e
+
+# This script is used to run the Playwright tests in a container
+# usage: ./docker_run.sh
+# Or to refresh the Cmfive Docker containers before running the tests
+# Node version can be specified, WARNING Playwright&Cosine will not be expected to support all node versions!
+# usage: ./docker_run.sh --fresh
+# usage: ./docker_run.sh --node_ver 18
+# usage: ./docker_run.sh --cosine_container [default is 'cmfive']
+
+freshen_all=""
+test_node_version=""
+cosine_container="cmfive"
+
+while test $# != 0
+do
+    case "$1" in
+    --fresh) freshen_all=true ;;
+    --node_ver)
+        shift; test_node_version=$1 ;;
+    --) shift; break;;
+    --cosine_container)
+        shift; cosine_container=$1 ;;
+    --) shift; break;;
+    esac
+    shift
+done
+
+# Bootstrap this script in the Playwright container
+if [ -z "$IS_PLAYWRIGHT_CONTAINER" ]; then
+
+
+    # Get the directory of the script
+    TESTDIR=$(dirname -- "$( readlink -f -- "$0"; )";)
+    
+    # Go up one directory from DIR until docker-compose.yml is found
+    PROJECTDIR=$TESTDIR
+    while [ ! -f "$PROJECTDIR/docker-compose.yml" ]; do
+        PROJECTDIR="$(dirname "$PROJECTDIR")"
+        echo "PROJECTDIR: $PROJECTDIR"
+        echo "TESTDIR: $TESTDIR"
+        if [ "$PROJECTDIR" = "/" ]; then
+            echo "Panic: docker-compose.yml not found, be sure to run this script from the test directory"
+            exit 2
+        fi
+    done
+
+    # if the --fresh flag is set, refresh the Cmfive Docker containers
+    if [ ! -z $freshen_all ]; then
+        echo "Refreshing Cmfive Docker containers"
+        docker compose down -v
+        docker compose up -d --wait
+    fi
+    
+    system_is_at=$(readlink -f $PROJECTDIR/system)
+    echo ">> CONTAINER LABELS (BOILERPLATE) <<"
+    echo "===================================="
+    docker inspect --format='{{range $key, $value := .Config.Labels}}{{$key}}={{$value}}{{println}}{{end}}' cmfive
+    echo ">> CONTAINER LABELS (CORE COMMIT) <<"
+    echo "===================================="
+    echo "VENDOR DIRECTORY:"
+    docker exec -u root $cosine_container sh -c "cd composer/vendor/2pisoftware/cmfive-core && git log -1 --pretty=format:"CORE_HASH=\"%H\"%nCORE_COMMIT_MSG=\"%s\"%nCORE_REF=\"%D\"""
+    echo ""
+    if cd $system_is_at/.. ; then
+        echo ""
+        echo ">> TESTS ON BRANCH (CORE COMMIT) <<"
+        echo "===================================="
+        echo "MOUNTED CORE from $system_is_at: (this is mounted to /system)"
+        git log -1 --pretty=format:"CORE_HASH=\"%H\"%nCORE_COMMIT_MSG=\"%s\"%nCORE_REF=\"%D\""
```

Comment:
```suggestion
    docker exec -u root $cosine_container sh -c "cd composer/vendor/2pisoftware/cmfive-core && git log --no-pager -1 --pretty=format:"CORE_HASH=\"%H\"%nCORE_COMMIT_MSG=\"%s\"%nCORE_REF=\"%D\"""
    echo ""
    if cd $system_is_at/.. ; then
        echo ""
        echo ">> TESTS ON BRANCH (CORE COMMIT) <<"
        echo "===================================="
        echo "MOUNTED CORE from $system_is_at: (this is mounted to /system)"
        git log --no-pager -1 --pretty=format:"CORE_HASH=\"%H\"%nCORE_COMMIT_MSG=\"%s\"%nCORE_REF=\"%D\""
```

Reply:
Needs a `--no-pager` flag so that interactive sessions (local) dont try to start a pager